activejob/lib/active_job/queue_adapters/sidekiq_adapter.rb

```diff
@@ -33,30 +33,34 @@ def enqueue_at(job, timestamp) # :nodoc:
       end
 
       def enqueue_all(jobs) # :nodoc:
+        enqueued_count = 0
         jobs.group_by(&:class).each do |job_class, same_class_jobs|
           same_class_jobs.group_by(&:queue_name).each do |queue, same_class_and_queue_jobs|
             immediate_jobs, scheduled_jobs = same_class_and_queue_jobs.partition { |job| job.scheduled_at.nil? }
 
             if immediate_jobs.any?
-              Sidekiq::Client.push_bulk(
+              jids = Sidekiq::Client.push_bulk(
                 "class" => JobWrapper,
                 "wrapped" => job_class,
                 "queue" => queue,
                 "args" => immediate_jobs.map { |job| [job.serialize] },
               )
+              enqueued_count += jids.count(&:present?)
             end
 
             if scheduled_jobs.any?
-              Sidekiq::Client.push_bulk(
+              jids = Sidekiq::Client.push_bulk(
                 "class" => JobWrapper,
                 "wrapped" => job_class,
                 "queue" => queue,
                 "args" => scheduled_jobs.map { |job| [job.serialize] },
                 "at" => scheduled_jobs.map { |job| job.scheduled_at }
               )
+              enqueued_count += jids.count(&:present?)
```

Comment:
Nitpick: what's the expected values of `jids` on insert failure? 

`present?` seem a bit weird in this context.

Reply:
Currently, sidekiq returns only real jids (job ids) of the jobs that were enqueued (so this array can be smaller in size than the number of jobs passed to `push_bulk`). So the `count(&:present?)` is not *yet* needed.

But in the upcoming sidekiq 7.1 it will return an array of exact same size as jobs passed to `push_bulk` (https://github.com/sidekiq/sidekiq/pull/5766) - some of the values will be jids, but some are `nil`s (for not enqueued jobs).

We can replace `select(&:present?)` by `reject(&:nil?)` if that helps.

Reply:
Right, that's what I suspected. If so, then `compact` will be the simpler method. No need to match a regexp against all `jids`.

Other than that, LGTM.

Reply:
Right 👍 Updated.